types/auth.ts

```diff
@@ -24,3 +24,7 @@ export interface SessionContextProps {
 export interface QueryParams {
     [key: string]: string | string[];
 }
+
+export interface QueryProps {
+    [key: string]: string; // might be able to expand QueryParams interface
```

Comment:
same with the comment here - this was only to explain the code in the comments. Also, this now looks largely similar to `QueryParams` - we can probably remove this and just point to `QueryParams` in hooks